src/app/components/student/student.component.ts

```diff
@@ -17,18 +22,22 @@ export class StudentComponent implements OnInit {
   deleteStudentId: string;
   sortOrder = '';
   clickedColumn: string;
+  filterSubject = new BehaviorSubject<{field: string, value: string}>({field: '', value: ''});
+  sortSubject = new BehaviorSubject<{sortField: string}>({sortField: ''});
+  isLoading = false;
+  currentFilterArray: Array<FilterField> = [];
 
   headElements = [
-    { title: '№', fieldName: 'number', sortable: false },
```

Comment:
'headElements' entity is getting larger, I would recommend to separate it to another file and use a const there, or place it within 'json' file.